### DISCUSSION TITLE
utf-8 nlohmann::detail::type_error

### DISCUSSION THREAD

**Discussion by 77h2l** [nlohmann/json#3221] [Q&A] [answered]
hi, I met a strange problem, my code was as follows,

  json InputParams;
  while (getline(fin, line)){
    std::string image_line = line;
    json a_config = generate_config_list(image_line.c_str());
    std::string img_path = a_config["filename"];
    a_config["image"] = parse_img_info(img_path.c_str());
    a_config["extend"] = "";
    InputParams.push_back(a_config);
    std::string InputParams_str = InputParams.dump();
    TimeWatch process_timer;
    process_timer.start();
    auto ret = CNCVOcrProcess(InputParams_str.c_str());
    LOG(INFO) << "process time cost : " << process_timer.stop() << " ms";
    ofs << ret << "\n";

 sometimes   std::string InputParams_str = InputParams.dump(); erroe report : what():  [json.exception.type_error.316] invalid UTF-8 byte at index 1547321: 0x9C

I have printed the jsonstring it was right,  how was that?

**Comment by nlohmann** [comment, marked as answer]
See https://json.nlohmann.me/home/exceptions/#jsonexceptiontype_error316 and https://json.nlohmann.me/api/basic_json/dump/#example

**Reply by 77h2l** [reply]
That was awesome right! Thx a lot!!
